python/jsbeautifier/javascript/beautifier.py

```diff
@@ -888,11 +909,10 @@ def handle_word(self, current_token):
                 prefix = 'NEWLINE'
 
         if reserved_array(current_token, ['else', 'catch', 'finally']):
-            if ((not (self._flags.last_token.type == TOKEN.END_BLOCK and self._previous_flags.mode == MODE.BlockStatement))
-                or self._options.brace_style == 'expand'
-                or self._options.brace_style == 'end-expand'
-                or (self._options.brace_style == 'none' and current_token.newlines)) \
-               and not self._flags.inline_frame:
+            blk_stmt_ends = self._flags.last_token.type == TOKEN.END_BLOCK
+            blk_stmt_ends = blk_stmt_ends and self._previous_flags.mode == MODE.BlockStatement
+            if ((not blk_stmt_ends) or self._options.brace_style in ['expand', 'end-expand', 'expand-all']
```

Comment:
New feature to have same behavior as that of brace style option with value expand or end-expand.